server/client.go

```diff
@@ -294,19 +294,25 @@ type pinfo struct {
 
 // outbound holds pending data for a socket.
 type outbound struct {
-	p   []byte        // Primary write buffer
-	s   []byte        // Secondary for use post flush
-	nb  net.Buffers   // net.Buffers for writev IO
-	sz  int32         // limit size per []byte, uses variable BufSize constants, start, min, max.
-	sws int32         // Number of short writes, used for dynamic resizing.
-	pb  int64         // Total pending/queued bytes.
-	pm  int32         // Total pending/queued messages.
-	fsp int32         // Flush signals that are pending per producer from readLoop's pcd.
-	sg  *sync.Cond    // To signal writeLoop that there is data to flush.
-	wdl time.Duration // Snapshot of write deadline.
-	mp  int64         // Snapshot of max pending for client.
-	lft time.Duration // Last flush time for Write.
-	stc chan struct{} // Stall chan we create to slow down producers on overrun, e.g. fan-in.
+	nb   net.Buffers   // Pending buffers for send, each has fixed capacity as per nbPool below.
+	wnb  net.Buffers   // Working copy of "nb", reused on each flushOutbound call, partial writes may leave entries here for next iteration.
+	orig net.Buffers   // Original copy of "nb", reused on each flushOutbound call.
+	pb   int64         // Total pending/queued bytes.
+	fsp  int32         // Flush signals that are pending per producer from readLoop's pcd.
+	sg   *sync.Cond    // To signal writeLoop that there is data to flush.
+	wdl  time.Duration // Snapshot of write deadline.
+	mp   int64         // Snapshot of max pending for client.
+	lft  time.Duration // Last flush time for Write.
+	stc  chan struct{} // Stall chan we create to slow down producers on overrun, e.g. fan-in.
+}
+
+const nbPoolSize = 512
+
+var nbPool = &sync.Pool{
```

Comment:
We most likely will need multiple pools. The dynamic sizing on blind reads is directly responsible for raw publish performance. This will greatly reduce.